File: src/main/java/dev/dediamondpro/minemark/elements/impl/text/TextElement.java
package dev.dediamondpro.minemark.elements.impl.text;

import dev.dediamondpro.minemark.LayoutConfig;
import dev.dediamondpro.minemark.elements.Element;
import org.jetbrains.annotations.Nullable;
import org.xml.sax.Attributes;

public abstract class TextElement<L extends LayoutConfig, R> extends AbstractTextElement<L, R> {

    public TextElement(@Nullable Element<L, R> parent, Attributes attributes) {
        super(parent, attributes);
    }

    @Override
    public String toString() {
        return text;
    }
}

File: elementa/src/main/kotlin/dev/dediamondpro/minemark/elementa/elements/TextElement.kt
package dev.dediamondpro.minemark.elementa.elements

import dev.dediamondpro.minemark.elementa.config.LayoutConfigImpl
import dev.dediamondpro.minemark.elementa.config.RenderConfigImpl
import dev.dediamondpro.minemark.elements.Element
import dev.dediamondpro.minemark.elements.impl.text.TextElement
import gg.essential.elementa.components.UIText
import gg.essential.elementa.dsl.constrain
import gg.essential.elementa.dsl.pixels
import org.xml.sax.Attributes

class TextElement(
    parent: Element<LayoutConfigImpl, RenderConfigImpl>?,
    attributes: Attributes?,
) : TextElement<LayoutConfigImpl, RenderConfigImpl>(parent, attributes) {
    private var element: UIText? = null

    override fun draw(x: Float, y: Float, width: Float, renderConfig: RenderConfigImpl) {
        if (element == null) element = UIText(text)
        element?.constrain {
            this.x = x.pixels()
            this.y = y.pixels()
        }
        element?.draw(renderConfig.matrixStack)
    }
}